lib/jellyfish/room_service.ex

```diff
@@ -66,8 +65,29 @@ defmodule Jellyfish.RoomService do
       )
     end
 
-    {min_node, _room_size} =
-      Enum.min_by(node_resources, fn {_node_name, room_num} -> room_num end)
+    {failed_rpcs, node_resources} =
+      Enum.split_with(node_resources, fn
+        {:badrpc, _info} -> true
+        _other -> false
+      end)
+
+    unless Enum.empty?(failed_rpcs) do
+      Logger.warn("These RPC calls fail: #{inspect(failed_rpcs)}")
+    end
+
+    %{node: min_node} =
+      Enum.min(
+        node_resources,
+        fn
+          %{forwarded_tracks_number: forwarded_tracks, rooms_number: rooms_num1},
+          %{forwarded_tracks_number: forwarded_tracks, rooms_number: rooms_num2} ->
+            rooms_num1 < rooms_num2
+
+          %{forwarded_tracks_number: forwarded_tracks1},
+          %{forwarded_tracks_number: forwarded_tracks2} ->
+            forwarded_tracks1 < forwarded_tracks2
+        end
+      )
```

Comment:
Let's put this in a separate function called e.g. `find_best_node` that will return single node